doc/demo.rst

```diff
@@ -210,108 +244,138 @@ Now we need to use some ``alpenhorn`` commands to create the Data Index
+    root@alpenshell:/#
 
 Once at the root prompt, we can build the data index and start
 populating it.
 
+.. tip::
+   You can log out of this ``alpenshell`` container at any time during the demo.  To later re-enter it,
+   simply run the ``docker compose run --rm alpenshell`` command again.
+
 Setting up the data index
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Creating the data index is simple, and can be accomplished with the alpenhorn CLI
-command:
+Creating the data index is simple, and can be accomplished by running the following
+command with the ``alpenhorn`` CLI utility:
 
-::
+.. code:: console
+   :class: democontainer
 
      alpenhorn db init
 
-Remember that all these alpenhorn commands need to be run inside the
-``alpen1`` container that we started in the last section. On successful completion,
-the ``db init`` command will report the version of the database schema used to create
-the Data Index:
+.. hint::
+   Remember that all these ``alpenhorn`` commands need to be run inside the
+   ``alpenshell`` container that we started in the last section.
+
+On successful completion, the ``db init`` command will report the version of the
+database schema used to create the Data Index:
 
-.. code::
+.. code:: console
+   :class: democontainer
 
-   root@alpen1:/# alpenhorn db init
+   root@alpenshell:/# alpenhorn db init
    Data Index version 2 initialised.
 
+.. tip::
+   It's worth pointing out at this point that the ``alpenhorn`` CLI can be run from
+   anywhere that has access to the alpenhorn database.  It's explicitly not necessary
+   to run the CLI on a host which contains a StorageNode (or is running the daemon),
+   even when using the CLI to run commands which affect that StorageNode or daemon.
+
 Setting up the import extension
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Because alpenhorn is data agnostic, it doesn’t have any facilities
+Because alpenhorn is data agnostic, it doesn't have any facilities
 out-of-the-box to import files. To be able to import files, alpenhorn
-needs one or more “import-detect extensions” to be loaded. For the
-purposes of this demo, we’ll use the simple ``pattern_importer`` example
+needs one or more "import-detect extensions" to be loaded. For the
+purposes of this demo, we'll use the simple ``pattern_importer`` example
 extension provided in the ``/examples`` directory. This extension has
-already been incorporated into the alpenhorn container image that we’re
+already been incorporated into the alpenhorn container image that we're
 running, and alpenhorn has been set up to use it.
 
+.. hint::
+   The reason alpenhorn is aware of the ``pattern_importer`` extension is
+   because it is listed as an extension to load in the alpenhorn config file,
+   which is available in the ``alpenshell`` at ``/etc/alpenhorn/alpenhorn.conf``.
+
+   You can also take a look at it on the docker host, in the ``/demo/``
+   subdirectory out of which you're running this demo.
+
```

Comment:
> -> Quick sentence to describe *how* the pattern_importer extension has been incorporated into the alpenhorn container image that we’re running and *how* alpenhorn has been set up to use it? Maybe point user to read demo_init function?

Here's a hint for that.  This is also reiterated below when the first daemon starts up and the user notices the `pattern_importer` extension being loaded.